kolibri/plugins/facility/assets/src/views/FacilityManagementSideNavEntry.vue

```diff
@@ -16,24 +32,132 @@
+  import { mapGetters } from 'vuex';
+  import { generateNavRoute } from '../../../../../core/assets/src/utils/generateNavRoutes';
+  import baseRoutes from '../baseRoutes';
+  import { PageNames as FacilityPageNames } from '../constants';
 
   const component = {
     name: 'FacilityManagementSideNavEntry',
     mixins: [commonCoreStrings],
     components: {
       CoreMenuOption,
     },
+    data() {
+      return {
+        visibleSubMenu: false,
+      };
+    },
+    mounted() {
+      this.submenuShouldBeOpen();
+    },
     computed: {
+      ...mapGetters(['isAppContext']),
       url() {
         return urls['kolibri:kolibri.plugins.facility:facility_management']();
       },
+      facilityRoutes() {
+        return {
+          facilityClasses: {
+            text: this.coreString('classesLabel'),
+            route: this.generateNavRoute(FacilityPageNames.CLASS_MGMT_PAGE),
+          },
+          facilityUsers: {
+            text: this.coreString('usersLabel'),
+            route: this.generateNavRoute(FacilityPageNames.USER_MGMT_PAGE),
+          },
+          facilitySettings: {
+            text: this.$tr('settingsLabel'),
+            route: this.generateNavRoute(FacilityPageNames.FACILITY_CONFIG_PAGE),
+          },
+          facilityData: {
+            text: this.$tr('data'),
+            route: this.generateNavRoute(FacilityPageNames.DATA_EXPORT_PAGE),
+          },
+        };
+      },
+      iconAfter() {
+        if (this.isAppContext) {
+          return this.visibleSubMenu ? 'chevronUp' : 'chevronDown';
+        }
+      },
+    },
+    methods: {
+      generateNavRoute(route) {
+        // if class id
+        let params;
+        if (this.$store.getters.currentFacilityId) {
+          params = { facilityId: this.$store.getters.currentFacilityId };
+          return generateNavRoute(this.url, route, baseRoutes, params);
+        }
+      },
+      toggleAndroidMenu() {
+        this.$emit('toggleAndroidMenu');
+      },
+      isActiveLink(route) {
+        return route.includes(this.$router.currentRoute.path);
+      },
+      submenuShouldBeOpen() {
+        // which plugin are we currently in?
+        this.visibleSubMenu = window.location.pathname.includes(this.url);
+      },
+      subpathStyles(route) {
+        if (this.isActiveLink(route)) {
+          return {
+            color: this.$themeTokens.primaryDark,
+            fontWeight: 'bold',
+            textDecoration: 'none',
+          };
+        }
+        return {
+          color: this.$themeTokens.text,
+          textDecoration: 'none',
+          ':hover': {
+            color: this.$themeTokens.primaryDark,
+            fontWeight: 'bold',
+          },
+          ':focus': this.$coreOutline,
+        };
+      },
+      handleNav(route) {
+        this.isActiveLink(route) ? this.toggleAndroidMenu() : null;
+      },
     },
     role: UserKinds.ADMIN,
     priority: 10,
+    $trs: {
+      data: {
+        message: 'Data',
```

Comment:
Presumably these are already defined somewhere else? Maybe just move them into a common translator object. https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/facility/assets/src/views/FacilityTopNav.vue#L54

Reply:
More new strings that have not been cleaned up.